### PR TITLE
Update RELEASE-NOTES.md

### PR DESCRIPTION
**Pull request by billpalombi**
Adds release notes

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,34 @@
 # Prefect Release Notes
 
+## Release 2.10.11
```
Comment (jakekaplan) [inline RELEASE-NOTES.md:3]: `2.10.10` maybe?

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,5 +1,34 @@
...
+
+### The Need for (CLI & import) Speed
+
+We wanted the CLI to be as fast as the rest of Prefect. Through a series of enhancements, we've sped up CLI performanace by as much as 4x on some systems! See the following pull requests for more information on how we sped up the CLI prefect import:
+- Delay `apprise` imports — https://github.com/PrefectHQ/prefect/pull/9557
+- Defer import of `dateparser` — https://github.com/PrefectHQ/prefect/pull/9582
+- Defer loading of prefect collections until necessary — https://github.com/PrefectHQ/prefect/pull/9571
```
Comment (discdiver) [inline RELEASE-NOTES.md:10]:
```suggestion
- Defer loading of Prefect integrations until necessary — https://github.com/PrefectHQ/prefect/pull/9571
```

Comment (discdiver) [reply RELEASE-NOTES.md:10]: We refer to them as integrations in the docs

Comment (billpalombi) [reply RELEASE-NOTES.md:10]: Gotcha

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,5 +1,34 @@
 # Prefect Release Notes
 
+## Release 2.10.10
+
+### The Need for (CLI & import) Speed
+
+We wanted the CLI to be as fast as the rest of Prefect. Through a series of enhancements, we've sped up CLI performanace by as much as 4x on some systems! See the following pull requests for more information on how we sped up the CLI prefect import:
```
Comment (serinamarie) [inline RELEASE-NOTES.md:7]:
```suggestion
We wanted the CLI to be as fast as the rest of Prefect. Through a series of enhancements, we've sped up CLI performance by as much as 4x on some systems! See the following pull requests for more information on how we sped up the CLI prefect import:
```

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,5 +1,34 @@
...
+### Fixes
+- Fix bug where `SIGTERM` was not properly captured as a flow run crash for flow runs created by a deployment — https://github.com/PrefectHQ/prefect/pull/9543
+- Fix deadlock when logging is overriden from an asynchronous context — https://github.com/PrefectHQ/prefect/pull/9602
+- Fix orchestration race conditions by adding lock for update to flow run state transitions — https://github.com/PrefectHQ/prefect/pull/9590
+
+### Documentation
+- Clarify the retry on tasks concept page — https://github.com/PrefectHQ/prefect/pull/9560
+- Improve the navigation structure and clarity of the API docs — https://github.com/PrefectHQ/prefect/pull/9574
+- Add `work_pool_name` to `Deployment.build_from_flow` on deployments concept page — https://github.com/PrefectHQ/prefect/pull/9581
+- Add Additional Worker Types to Work Pools, Workers & Agents Concept page — https://github.com/PrefectHQ/prefect/pull/9580
```
Comment (serinamarie) [inline RELEASE-NOTES.md:26]:
```suggestion
- Add additional worker types to work pools, workers & agents concept page — https://github.com/PrefectHQ/prefect/pull/9580
```

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,5 +1,34 @@
...
+- Add docstrings for all schema filters — https://github.com/PrefectHQ/prefect/pull/9572
+
+## New Contributors
+- @seanpwlms made their first contribution in https://github.com/PrefectHQ/prefect/pull/9608
```
Comment (serinamarie) [inline RELEASE-NOTES.md:30]: internal to Prefect
```suggestion
```

File: RELEASE-NOTES.md [hunk 6]
```diff
@@ -1,5 +1,34 @@
...
+
```
Comment (serinamarie) [inline RELEASE-NOTES.md:31]:
```suggestion

**All changes**: https://github.com/PrefectHQ/prefect/compare/2.10.9...2.10.10
```

File: RELEASE-NOTES.md [hunk 7]
```diff
@@ -1,5 +1,42 @@
...
+### The Need for (CLI & import) Speed
+
+We wanted the CLI to be as fast as the rest of Prefect. Through a series of enhancements, we've sped up CLI performance by as much as 4x on some systems! See the following pull requests for more information on how we sped up the CLI prefect import:
+- Delay `apprise` imports — https://github.com/PrefectHQ/prefect/pull/9557
+- Defer import of `dateparser` — https://github.com/PrefectHQ/prefect/pull/9582
+- Defer loading of Prefect integrations until necessary — https://github.com/PrefectHQ/prefect/pull/9571
+- Add `Block.get_block_class_from_key` and replace external uses of `lookup_type` — https://github.com/PrefectHQ/prefect/pull/9621
+
+### Enhancements
+- Add SIGTERM handling for workers — https://github.com/PrefectHQ/prefect/pull/9530
```
Comment (serinamarie) [inline RELEASE-NOTES.md:14]:
```suggestion
- Handle SIGTERM received by worker gracefully — https://github.com/PrefectHQ/prefect/pull/9530
```

File: RELEASE-NOTES.md [hunk 8]
```diff
@@ -1,5 +1,42 @@
...
+- Add ability to view table artifacts with NaN values in the Prefect UI — https://github.com/PrefectHQ/prefect/pull/9585
+- Update `prefect version` command to avoid creating the database if it does not exist — https://github.com/PrefectHQ/prefect/pull/9586
+- Use "eager" upgrade strategy when upgrading requirements in Docker images — https://github.com/PrefectHQ/prefect/pull/9640
+- Allow client retries when server SQLite database is busy — https://github.com/PrefectHQ/prefect/pull/9632
+- Add lock for update to flow run state transitions — https://github.com/PrefectHQ/prefect/pull/9590
+
+### Fixes
+- Fix bug where `SIGTERM` was not properly captured as a flow run crash for flow runs created by a deployment — https://github.com/PrefectHQ/prefect/pull/9543
+- Fix deadlock when logging is overriden from an asynchronous context — https://github.com/PrefectHQ/prefect/pull/9602
+- Fix orchestration race conditions by adding lock for update to flow run state transitions — https://github.com/PrefectHQ/prefect/pull/9590
```
Comment (serinamarie) [inline RELEASE-NOTES.md:24]: There are a few PRs related to orchestration race conditions 🤔 Wonder if that's worth mentioning in one place

File: RELEASE-NOTES.md [hunk 9]
```diff
@@ -1,5 +1,42 @@
...
+- Load collections before auto-registering block types on the server — https://github.com/PrefectHQ/prefect/pull/9626
+- Date Range filter not responsive on flow runs page — https://github.com/PrefectHQ/prefect/pull/9636
```
Comment (serinamarie) [inline RELEASE-NOTES.md:26]:
```suggestion
- Fix date range filter on flow runs page — https://github.com/PrefectHQ/prefect/pull/9636
```

File: RELEASE-NOTES.md [hunk 10]
```diff
@@ -1,5 +1,42 @@
...
+- Remove image layer caching during builds — https://github.com/PrefectHQ/prefect/pull/9587
```
Comment (serinamarie) [inline RELEASE-NOTES.md:27]: Is this helpful to the user? Otherwise not sure if it needs mentioning

File: RELEASE-NOTES.md [hunk 11]
```diff
@@ -1,5 +1,42 @@
...
+- Update `prefect version` command to avoid creating the database if it does not exist — https://github.com/PrefectHQ/prefect/pull/9586
```
Comment (serinamarie) [inline RELEASE-NOTES.md:28]: This PR is listed under enhancements as well.

File: RELEASE-NOTES.md [hunk 12]
```diff
@@ -1,5 +1,42 @@
...
+
+### Documentation
+- Clarify the retry on tasks concept page — https://github.com/PrefectHQ/prefect/pull/9560
+- Improve the navigation structure and clarity of the API docs — https://github.com/PrefectHQ/prefect/pull/9574
+- Add `work_pool_name` to `Deployment.build_from_flow` on deployments concept page — https://github.com/PrefectHQ/prefect/pull/9581
+- Add additional worker types to work pools, workers & agents concept page — https://github.com/PrefectHQ/prefect/pull/9580
+- Add docstrings for all schema filters — https://github.com/PrefectHQ/prefect/pull/9572
+
+## New Contributors
```
Comment (serinamarie) [inline RELEASE-NOTES.md:37]: No new contributors so this can probably be removed

File: RELEASE-NOTES.md (hunk 10 above)
Comment (jakekaplan) [reply RELEASE-NOTES.md:27]: great point!

File: RELEASE-NOTES.md (hunk 7 above)
Comment (serinamarie) [reply RELEASE-NOTES.md:14]: this is the wording we've used for the agent and server

File: RELEASE-NOTES.md [hunk 13]
```diff
@@ -1,5 +1,42 @@
...
+- Update `prefect version` command to avoid creating the database if it does not exist — https://github.com/PrefectHQ/prefect/pull/9586
+- Use "eager" upgrade strategy when upgrading requirements in Docker images — https://github.com/PrefectHQ/prefect/pull/9640
+- Allow client retries when server SQLite database is busy — https://github.com/PrefectHQ/prefect/pull/9632
+- Add lock for update to flow run state transitions — https://github.com/PrefectHQ/prefect/pull/9590
+
+### Fixes
+- Fix bug where `SIGTERM` was not properly captured as a flow run crash for flow runs created by a deployment — https://github.com/PrefectHQ/prefect/pull/9543
+- Fix deadlock when logging is overriden from an asynchronous context — https://github.com/PrefectHQ/prefect/pull/9602
+- Fix orchestration race conditions by adding lock for update to flow run state transitions — https://github.com/PrefectHQ/prefect/pull/9590
+- Load collections before auto-registering block types on the server — https://github.com/PrefectHQ/prefect/pull/9626
```
[the remark this answers is not in the document]
Comment (jakekaplan) [reply RELEASE-NOTES.md:25]: I think it's related follow up, I'll take it out

File: RELEASE-NOTES.md (hunk 7 above)
Comment (serinamarie) [reply RELEASE-NOTES.md:14]: 🦢

File: RELEASE-NOTES.md (hunk 8 above)
Comment (jakekaplan) [reply RELEASE-NOTES.md:24]: ah I think I duplicated one! removed!

File: RELEASE-NOTES.md (hunk 11 above)
Comment (jakekaplan) [reply RELEASE-NOTES.md]: 👍

File: RELEASE-NOTES.md [hunk 14]
```diff
@@ -1,5 +1,36 @@
 # Prefect Release Notes
 
+## Release 2.10.10
+
+### The Need for (CLI & import) Speed
```
Comment (serinamarie) [inline RELEASE-NOTES.md:5]: We tend to not capitalize the full heading
```suggestion
### The need for (CLI & import) speed
```

**Review by serinamarie — APPROVED** [review]

**Pull request merged**